apps/server/src/modules/group/uc/group.uc.ts

```diff
@@ -377,13 +85,21 @@ export class GroupUc {
 
 		const canSeeFullList: boolean = this.authorizationService.hasAllPermissions(user, [Permission.GROUP_FULL_ADMIN]);
 
-		let groups: Page<Group>;
+		const filter: IGroupFilter = { nameQuery };
+		options.order = { name: SortOrder.asc };
+
 		if (canSeeFullList) {
-			groups = await this.getGroupsForSchool(school, query, availableGroupsForCourseSync);
+			filter.schoolId = schoolId;
 		} else {
-			groups = await this.getGroupsForUser(userId, query, availableGroupsForCourseSync);
+			filter.userId = userId;
 		}
 
+		if (this.configService.get('FEATURE_SCHULCONNEX_COURSE_SYNC_ENABLED')) {
+			filter.availableGroupsForCourseSync = availableGroupsForCourseSync;
```

Comment:
The `availableGroupsForCourseSync` is not used when you call `findAvailableGroups` or `findGroups`, but they still use the `IGroupFilter`-Interface that contains this attribute. I know I wrote this line as an example in my comment, but I also  did not distinguish between the two separate service calls and it does not really make sense to include `availableGroupsForCourseSync` in the `IGroupFilter`-Interface, when the service also uses the Interface. The `availableGroupsForCourseSync && this.configService.get('FEATURE_SCHULCONNEX_COURSE_SYNC_ENABLED')` was better for type safety 🙂.

Reply:
will revert that ^^